### PR TITLE
a left over from the 8-bit world.

### PR DESCRIPTION
**Pull request by robert-b**
now the reference vector is closer to where it should be.

### CODE REVIEW

**Comment by Quick-Flash** [comment]
Intersting, emuboost but only having effect near center stick and that effect slows down as you move your stick. If this is the effect you are going for you might want to try setpoint instead of stick deflection as that will have a more uniform feel and tune across different rates.

**Comment by hydra** [comment]
@robert-b Can you update the title and description with more details on the proposed change please.

**Comment by robert-b** [comment]
the title is correct.
in the 8bit world, aka multiwii, float's where expensive.
nowadays summing up float's using an int is bad practice.
not clean at all.
and when looking at the gaussian noise distribution, the reference vector for the sensor fusion might get screwed up.
https://github.com/multiwii/baseflight/blob/e582e0ce6321b7dd2293a60264e1040d0cbac5a4/src/sensors.c#L276
https://github.com/multiwii/multiwii-firmware/blob/e957a701844a9ecf0d19ac6a7e3ea1e8d81cc43b/Sensors.cpp#L232

those converting the sources it made a bad job ;-)

**Comment by robert-b** [comment]
the scope of the pr should only be:
https://github.com/cleanflight/cleanflight/pull/3105/commits/bcd22df62d66ed75cd432be7ed5050b422690943
i overlooked the dynamic view of github.

**Comment by hydra** [comment]
@robert-b ahh, so the original port from Multiwii (8-bit) to the F1 (32-bit) didn't use floats because the F1 doesn't have an FPU, so it's not so much a left-over as you might think.  Only since the F3 have we had an FPU.  It's more of a left-over from the F1 days 😄 

A better title, so that others can follow better, might have been 'Use `floats` for accelerometer calibration for better precision'.  Can you create a new PR or update this PR's branch with just https://github.com/cleanflight/cleanflight/commit/bcd22df62d66ed75cd432be7ed5050b422690943 please?

**Comment by robert-b** [comment]
hmm, the availabilty of floats is not fpu dependent.
with an fpu the code is executed slightly faster.
even on an stm32f1 the code is wrong.
i am off for the next few days ...

**Comment by hydra** [comment]
@robert-b yes, it was partly that, and partly that using floats means you have to include math libraries, which has an impact on the flash size.

Please make the PR to fix when you can.

File: src/main/sensors/acceleration.c [hunk 1]
```diff
@@ -424,9 +424,9 @@ static void performAcclerationCalibration(rollAndPitchTrims_t *rollAndPitchTrims
 
     if (isOnFinalAccelerationCalibrationCycle()) {
         // Calculate average, shift Z down by acc_1G and store values in EEPROM at end of calibration
-        accelerationTrims->raw[X] = (a[X] + (CALIBRATING_ACC_CYCLES / 2)) / CALIBRATING_ACC_CYCLES;
-        accelerationTrims->raw[Y] = (a[Y] + (CALIBRATING_ACC_CYCLES / 2)) / CALIBRATING_ACC_CYCLES;
-        accelerationTrims->raw[Z] = (a[Z] + (CALIBRATING_ACC_CYCLES / 2)) / CALIBRATING_ACC_CYCLES - acc.dev.acc_1G;
+        accelerationTrims->raw[X] = lrintf((a[X] + (CALIBRATING_ACC_CYCLES / 2)) / CALIBRATING_ACC_CYCLES);
```
Comment (ledvinap) [inline src/main/sensors/acceleration.c:427]: Old code did round almost correctly (floor(x + .5) == round(x), but integer division is `trunc()`, so it works only for positive values), now it's offset by +0.5
Also both methods are IMO identical (except for negative value) - accADC is integer, so sum is exact and  division will produce identical result

Comment (hydra) [reply src/main/sensors/acceleration.c:427]: sounds like a good case for a unit test.

**Pull request closed without merging**